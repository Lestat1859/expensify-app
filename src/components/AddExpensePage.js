import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense); // le props onSubmit executera le code présent dans mapDispatchToProps
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}


const mapDispatchToProps = (dispatch) => ( // on passe dispatch comme argument pour pouvoir l'utiliser
    {addExpense : (expense) => dispatch(addExpense(expense))} //la propriété onSubmit va appeller dispacth de addExpense avec expense en paramétre 
);

export default connect(undefined,mapDispatchToProps)(AddExpensePage); // Ici on passe comme premier argument undefined car on a pas besoin de récupérer le contenu du store.


