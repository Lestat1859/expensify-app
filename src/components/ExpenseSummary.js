//1 Create ExpenseSummary Component

// render by dashboardPage
// Test with two snapshot tests
// connect to the store for :
//- expenseCount (how many visible expenses ?)
// - expenseTotal (what's the total of the visible expense ?)

// Example
// Viewing 2 expenses totalling $94.34

//Example
// Viewing 1 expense totalling $94.34

//2 Commit and deploy

import React from 'react';
import { connect } from 'react-redux';
import selectExpenseTotal from '../selectors/expenses-total';
import numeral from 'numeral';

class ExpenseSummary extends React.Component {

    render(){
        return (
        <div>
            {this.props.expenseCount >0 ? <p>Viewing {this.props.expenseCount} expense(s) totaling {this.props.expenseTotal}</p> : '' }
        </div>
        )
    }
};
const mapStateToProps = (state) => ({
    expenseCount: state.expenses.length,
    expenseTotal : numeral(selectExpenseTotal(state.expenses)/100).format('$0,0.00')
  });
  

export default connect(mapStateToProps)(ExpenseSummary);