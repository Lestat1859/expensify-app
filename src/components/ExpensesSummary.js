//1 Create ExpenseSummary Component

// render by dashboardPage
// Test with two snapshot tests
// connect to the store for :
//- expenseCount (how many visible expenses ?)
// - expenseTotal (what's the total of the visible expense ?)

// Example
// Viewing 2 expenses totalling $94.34

//Example
// Viewing 1 expense totalling $94.34

//2 Commit and deploy

import React from 'react';
import { connect } from 'react-redux';
import selectExpenseTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expenseTotal})=>{
    return (
    <div>
        {expenseCount >0 ? <h2>Viewing {expenseCount} expense(s) totaling {expenseTotal}</h2> : '' }
    </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses,state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal : numeral(selectExpenseTotal(visibleExpenses)/100).format('$0,0.00')
    }    
};  

export default connect(mapStateToProps)(ExpensesSummary);