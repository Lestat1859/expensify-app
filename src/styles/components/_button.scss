.button {
  background-color: $blue;
  border: none;
  color: white;
  display:inline-block;
  font-weight: 300;
  font-size: $font-size-large;
  padding : $s-size;  
  line-height: 1;
  text-decoration: none;

}

.button--link{
  background: none;
}

.button--secondary{
  background: #EA4C36;
}