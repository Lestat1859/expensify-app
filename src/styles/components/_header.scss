.header {
  background: $dark-blue;
}

.header__content{
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header__title{
  color:white;
  text-decoration: none; // enlève le souligné du lien
  h1 {
    margin :0; //supprime le blanc en haut
  }
}