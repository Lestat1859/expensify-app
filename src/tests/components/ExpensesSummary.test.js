import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render ExpenseSummary correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={15} />);
  expect(wrapper).toMatchSnapshot();
});


test('should render ExpenseSummary correctly with 2 expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={30} />);
  expect(wrapper).toMatchSnapshot();
});