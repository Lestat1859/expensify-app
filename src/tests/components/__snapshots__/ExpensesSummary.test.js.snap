// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 2 expenses 1`] = `
<div>
  <h2>
    Viewing 
    2
     expense(s) totaling 
    30
  </h2>
</div>
`;

exports[`should render ExpenseSummary correctly with one expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     expense(s) totaling 
    15
  </h2>
</div>
`;
